schemas/metadata-schema.json

```
@@ -0,0 +1,402 @@
+          },
+          "countmax": {
+            "type": "number",
+            "description": "Maximum number of devices required"
+          }
+        }
+      }
+    },
+    "base": {
+      "type": "object",
+      "description": "A list of bases in descending order of preference for use in resolving a container image. Must not be present if resource is specified. These bases are listed as base (instead of name) and channel as in the Base definition, as an unnamed top-level object list.",
+      "additionalProperties": {
+        "type": "object",
+        "properties": {
+          "name": {
+            "type": "string",
+            "description": "Name of the OS e.g. ubuntu, centos, windows, osx, opensuse or genericlinux."
+          },
+          "channel": {
+            "type": "string",
+            "description": "Channel of the OS (track/risk/branch) such as used for Snaps e.g. 18.04/stable or 20.04/stable/fips."
+          },
+          "architectures": {
+            "type": "array",
+            "description": "List of architectures that this particular charm build can run on.",
+            "items": {
+              "type": "string"
+            }
+          } 
+        }
+      }
+    },
+    "mount": {
+      "type": "array",
+      "description": "List of mounted storage for this container",
+      "items": {
+        "type": "object",
+        "properties": {
+          "storage": {
+            "type": "string",
+            "description": "Name of storage to mount from the charm storage."
+          },
+          "location": {
+            "type": "string",
+            "description": "Location is the mount location for filesystem stores. For multi-stores, the location acts as the parent directory for each mounted store."
+          }
+        },
+        "required": [ "storage" ]
+      }
+    },
+    "container": {
+      "type": "object",
+      "description": "Map of containers to be created adjacent to the charm. On kubernetes these are sidecars. Other platforms the behaviour is to be determined and will likely be fulfilled by a container runtime like LXD, containerd, Docker. Adjacency control is likely to arrive in a future revision (such as charm running on machine-0 and container-1 running on machine-1 etc.). The key represents the container name.",
+      "additionalProperties": {
+        "type": "object",
+        "properties": {
+          "resource": {
+            "type": "string",
+            "description": "Reference for on entry in the resources field, indicating a specific container image. Must not be present if a base/channel is specified"
+          },
+          "bases": {
+            "$ref": "#/definitions/base"
+          },
+          "mounts": {
+            "$ref": "#/definitions/mount"
+          }
+        }
+      }
+    },
+    "deploymentType": {
+      "type": "string",
+      "description": "Type of deployment in kubernetes",
```

nit: Mixed capitalization of Kubernetes in descriptions

```suggestion
      "description": "Type of deployment in Kubernetes",
```